File: frontend/src/api/modules/nginx.ts
import http from '@/api/http'

// --- Nginx 状态 ---
export const getNginxStatus = () => {
  return http.get('/nginx/status')
}

// --- Nginx 操作（start/stop/reload/reopen/quit）---
export const operateNginx = (operation: string) => {
  return http.post('/nginx/operate', { operation })
}

// --- 配置测试 ---
export const testNginxConfig = () => {
  return http.get('/nginx/config-test')
}

// --- 安装 Nginx ---
export const installNginx = (method: string, version?: string) => {
  return http.post('/nginx/install', { method, version })
}

// --- 安装进度 ---
export const getInstallProgress = () => {
  return http.get('/nginx/install/progress')
}

// --- 卸载 Nginx ---
export const uninstallNginx = (forceCleanup = false, mode?: string) => {
  return http.post('/nginx/uninstall', { forceCleanup, mode })
}

// --- 获取可用的预编译版本列表 ---
export const listNginxVersions = () => {
  return http.get('/nginx/versions')
}

// --- 检查 Nginx 更新 ---
export const checkNginxUpdate = () => {
  return http.get('/nginx/update/check')
}

// --- 升级 Nginx ---
export const upgradeNginx = (version?: string) => {
  return http.post('/nginx/update/upgrade', { version })
}

// --- 设置 Nginx 开机自启 ---
export const setNginxAutoStart = (enable: boolean) => {
  return http.post('/nginx/autostart', { enable })
}

File: backend/app/api/v1/nginx.go
package v1

import (
	"net/http"

	"xpanel/app/api/v1/helper"
	"xpanel/app/dto"
	"xpanel/app/service"

	"github.com/gin-gonic/gin"
)

type NginxAPI struct{}

var nginxService = service.NewINginxService()
var nginxInstallService = service.NewINginxInstallService()

// GetNginxStatus 获取 Nginx 运行状态
func (api *NginxAPI) GetNginxStatus(c *gin.Context) {
	status, err := nginxService.GetStatus()
	if err != nil {
		helper.ErrorWithDetail(c, http.StatusInternalServerError, err.Error())
		return
	}
	helper.SuccessWithData(c, status)
}

// OperateNginx 执行 Nginx 操作（start/stop/reload/reopen/quit）
func (api *NginxAPI) OperateNginx(c *gin.Context) {
	var req dto.NginxOperateReq
	if err := helper.CheckBindAndValidate(&req, c); err != nil {
		return
	}
	if err := nginxService.Operate(req); err != nil {
		helper.ErrorWithDetail(c, http.StatusInternalServerError, err.Error())
		return
	}
	helper.SuccessWithData(c, nil)
}

// TestNginxConfig 测试 Nginx 配置
func (api *NginxAPI) TestNginxConfig(c *gin.Context) {
	result, err := nginxService.TestConfig()
	if err != nil {
		helper.ErrorWithDetail(c, http.StatusInternalServerError, err.Error())
		return
	}
	helper.SuccessWithData(c, result)
}

// InstallNginx 安装 Nginx（apt 或预编译）
func (api *NginxAPI) InstallNginx(c *gin.Context) {
	var req dto.NginxInstallReq
	if err := c.ShouldBindJSON(&req); err != nil {
		helper.ErrorWithDetail(c, http.StatusBadRequest, err.Error())
		return
	}
	if err := nginxInstallService.Install(req); err != nil {
		helper.ErrorWithDetail(c, http.StatusInternalServerError, err.Error())
		return
	}
	helper.SuccessWithData(c, nil)
}

// GetInstallProgress 获取 Nginx 安装进度
func (api *NginxAPI) GetInstallProgress(c *gin.Context) {
	progress := nginxInstallService.GetProgress()
	helper.SuccessWithData(c, progress)
}

// UninstallNginx 卸载 Nginx
func (api *NginxAPI) UninstallNginx(c *gin.Context) {
	var req dto.NginxUninstallReq
	_ = c.ShouldBindJSON(&req)
	if err := nginxInstallService.Uninstall(req); err != nil {
		helper.ErrorWithDetail(c, http.StatusInternalServerError, err.Error())
		return
	}
	helper.SuccessWithData(c, nil)
}

// SetNginxAutoStart 设置 Nginx 开机自启
func (api *NginxAPI) SetNginxAutoStart(c *gin.Context) {
	var req dto.NginxAutoStartReq
	if err := helper.CheckBindAndValidate(&req, c); err != nil {
		return
	}
	if err := nginxService.SetAutoStart(req.Enable); err != nil {
		helper.ErrorWithDetail(c, http.StatusInternalServerError, err.Error())
		return
	}
	helper.SuccessWithData(c, nil)
}

// ListNginxVersions 获取可用的 Nginx 预编译版本列表
func (api *NginxAPI) ListNginxVersions(c *gin.Context) {
	versions, err := nginxInstallService.ListVersions()
	if err != nil {
		helper.ErrorWithDetail(c, http.StatusInternalServerError, err.Error())
		return
	}
	helper.SuccessWithData(c, versions)
}

// CheckNginxUpdate 检查 Nginx 是否有可用更新
func (api *NginxAPI) CheckNginxUpdate(c *gin.Context) {
	info, err := nginxInstallService.CheckUpdate()
	if err != nil {
		helper.ErrorWithDetail(c, http.StatusInternalServerError, err.Error())
		return
	}
	helper.SuccessWithData(c, info)
}

// UpgradeNginx 升级 Nginx
func (api *NginxAPI) UpgradeNginx(c *gin.Context) {
	var req dto.NginxUpgradeReq
	_ = c.ShouldBindJSON(&req)
	if err := nginxInstallService.Upgrade(req); err != nil {
		helper.ErrorWithDetail(c, http.StatusInternalServerError, err.Error())
		return
	}
	helper.SuccessWithData(c, nil)
}
